lib/services/history/history_manager.ts

```diff
@@ -120,9 +133,18 @@ export class SNHistoryManager extends PureService {
     this.historySession!.setItemRevisionThreshold(threshold);
   }
 
-  async addHistoryEntryForItem(item: SNItem) {
-    const payload = CreateSourcedPayloadFromObject(item, PayloadSource.SessionHistory)
-    const entry = this.historySession!.addEntryForPayload(payload);
+  addEntryForPayload(payload: PurePayload) {
+    switch (payload.source) {
+      case PayloadSource.ServerHistory:
+        return this.historyServer!.addEntryForPayload(payload);
```

Comment:
I don't think this is necessary. We should just return to the client an array of revisions. No need for this encapsulation. The history server objet should not exist.